pkg/core/xds/sockets.go

```diff
@@ -5,8 +5,6 @@ import (
 	"path/filepath"
 )
 
-// TODO inline these in kuma-dp once backward compatibility is removed. https://github.com/kumahq/kuma/issues/7220
-
 // AccessLogSocketName generates a socket path that will fit the Unix socket path limitation of 104 chars
 func AccessLogSocketName(tmpDir, name, mesh string) string {
 	return socketName(filepath.Join(tmpDir, fmt.Sprintf("kuma-al-%s-%s", name, mesh)))
```

Comment:
Since this function relies on a private function `socketName` and there are also some other functions in this file, maybe we could just keep `AccessLogSocketName` here?

Reply:
Had an offline discussion with @Icarus9913 , I think if the `socketName` is also a dependent of other functions and those can not be inlined togher, we may only move this single function `AccessLogSocketName` to kuma-dp, and make `socketName` public.

Reply:
**All of these** should be moved to kuma-dp and inlined if that makes sense. The core problem is that before some of the paths were generated in the CP that can have different paths than the DP. The correct way of doing things is generating these on `DP` and sending it in metadata.

Reply:
ok, looking at this again, the first two can actually be moved to kuma-dp, the third one is used both on kuma-dp (to scrape config) and kuma-cp (to generate Envoy listener) and the forth one only in kuma-dp. So I guess only the third one needs to be shared and there is no workaround. So I think we can make the function public and move whatever we can.